src/v/pandaproxy/rest/handlers.cc

```diff
@@ -175,17 +204,9 @@ get_topics_records(server::request_t rq, server::reply_t rp) {
       max_bytes);
 
     return rq.service()
-      .client()
-      .local()
-      .fetch_partition(std::move(tp), offset, max_bytes, timeout)
-      .then([res_fmt, rp = std::move(rp)](kafka::fetch_response res) mutable {
-          ::json::StringBuffer str_buf;
-          ::json::Writer<::json::StringBuffer> w(str_buf);
-
-          ppj::rjson_serialize_fmt(res_fmt)(w, std::move(res));
-
-          // TODO Ben: Prevent this linearization
-          ss::sstring json_rslt = str_buf.GetString();
+      .client_cache()
+      .invoke_on_cache(rq.user, handler)
+      .then([res_fmt, rp = std::move(rp)](ss::sstring json_rslt) mutable {
```

Comment:
Had to keep a continuation here because moving the `server::reply_t` into the handler causes a compile error due to deleted copy constructor. The compile error is because seastar perfect-forwards arguments in `sharded.invoke_on` and apparently that forwarding invokes the (deleted) copy constructor on the unique_ptr within `sever::reply_t`.

Reply:
that's ok. did you consider leaving the json string handling in the continuation like before?

Reply:
Tested it out and the compiler doesn't like returning the `str_buf` in a future. It violates the nothrow_move_constructible requirement.

```
/home/nyalia/redpanda/vbuild/debug/clang/rp_deps_install/include/seastar/core/future.hh:591:5: error: static_assert failed due to requirement 'std::is_nothrow_move_constructible<rapidjson::GenericStringBuffer<rapidjson::UTF8<char>, json::throwing_allocator>>::value' "Types must be no-throw move constructible"
```